src/core_ocean/mode_forward/mpas_ocn_time_integration_split.F

```diff
@@ -1477,712 +1492,744 @@ subroutine ocn_time_integrator_split(domain, dt)!{{{
-               ! in this diagnostics solve.
-               call ocn_diagnostic_solve(dt, statePool, forcingPool, meshPool, &
-                          scratchPool, tracersPool, 2, full=.false.)
+            !$omp parallel
+            !$omp do schedule(runtime) private(k)
+            do iCell = 1, nCellsAll
+            do k = 1, maxLevelCell(iCell)
+               ! this is h_{n+1}
+               layerThicknessNew(k,iCell) = &
+                                     layerThicknessCur(k,iCell) + &
+                                  dt*layerThicknessTend(k,iCell)
+            end do
+            end do
+            !$omp end do
+            !$omp end parallel
 
-            !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-            !
-            !  If large iteration complete, compute all variables at time n+1
-            !
-            !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-            elseif (split_explicit_step == config_n_ts_iter) then
+            if (config_compute_active_tracer_budgets) then
 
                !$omp parallel
                !$omp do schedule(runtime) private(k)
-               do iCell = 1, nCells
-                  do k = 1, maxLevelCell(iCell)
-                     ! this is h_{n+1}
-                     layerThicknessNew(k,iCell) = layerThicknessCur(k,iCell) + dt * layerThicknessTend(k,iCell)
-                  end do
-               end do
+               do iEdge = 1, nEdgesAll
+               do k= 1, maxLevelEdgeTop(iEdge)
+                  activeTracerHorizontalAdvectionEdgeFlux(:,k,iEdge) = &
+                  activeTracerHorizontalAdvectionEdgeFlux(:,k,iEdge) / &
+                       layerThickEdge(k,iEdge)
+               enddo
+               enddo
                !$omp end do
-               !$omp end parallel
-
-               if (config_compute_active_tracer_budgets) then
-                  !$omp parallel
-                  !$omp do schedule(runtime) private(k)
-                  do iEdge = 1, nEdges
-                     do k= 1, maxLevelEdgeTop(iEdge)
-                        activeTracerHorizontalAdvectionEdgeFlux(:,k,iEdge) = &
-                          activeTracerHorizontalAdvectionEdgeFlux(:,k,iEdge) / &
-                          layerThickEdge(k,iEdge)
-                     enddo
-                  enddo
-                  !$omp end do
 
-                  !$omp do schedule(runtime) private(k)
-                  do iCell = 1, nCells
-                     do k= 1, maxLevelCell(iCell)
-                        activeTracerHorizontalAdvectionTendency(:,k,iCell) = &
-                           activeTracerHorizontalAdvectionTendency(:,k,iCell) / &
-                           layerThicknessNew(k,iCell)
+               !$omp do schedule(runtime) private(k)
+               do iCell = 1, nCellsAll
+               do k= 1, maxLevelCell(iCell)
+                  activeTracerHorizontalAdvectionTendency(:,k,iCell) = &
```

Comment:
@mattdturner or @philipwjones These tendency variables are on a per-tracer basis within the active Tracers. I think we need to still use the `get_array` command within a tracer loop, rather than define it in `mpas_ocn_diagnostics_variables.F`. I think this mistake was added with the diagnostics variables PR a month ago.

Reply:
Never mind. I just answered my own question. The first index is over tracers, so it can indeed be defined in `mpas_ocn_diagnostics_variables.F`.